lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -297,7 +298,12 @@ public boolean isRetryDelayValid()
     @AssertTrue(message = "Either gcs.auth-type or gcs.json-key or gcs.json-key-file-path must be set")
     public boolean isAuthMethodValid()
     {
-        if (getAuthType() == AuthType.ACCESS_TOKEN) {
+        // allow not specify any of configuration for SERVICE_ACCOUNT auth
+        if (getAuthType() == AuthType.SERVICE_ACCOUNT && jsonKey == null && jsonKeyFilePath == null) {
```

Comment:
We could bound separate config based on a type to have these checked for other then ASSUMED methods